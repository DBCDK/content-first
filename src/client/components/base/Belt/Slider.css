.belts {
  padding-right: 30px;
}

.desktop-slider .swiper-wrapper {
  padding: var(--6pt) var(--5pt) 0 var(--5pt);
}

.childBelt .desktop-slider .swiper-wrapper {
  padding: var(--5pt);
}

.swiper-wrapper .WorkCard {
  margin: 0 var(--3pt) 0 0;
}

.swiper-wrapper .WorkCard:last-child {
  margin: 0 var(--5pt) 0 0;
}

.mobile-slider {
  padding: 0 var(--3pt);
}

.mobile-slider .WorkCard {
  margin: 0 var(--2pt) 0 0;
}

.mobile-slider .WorkCard:last-child {
  margin: 0 var(--3pt) 0 0;
}

@media only screen and (max-width: 576px) {
  .mobile-slider {
    padding: 0 var(--2pt);
  }

  .mobile-slider .WorkCard:last-child {
    margin: 0 var(--2pt) 0 0;
  }
}

.desktop-slider .swiper-container {
  padding-right: 42px;
}

.desktop-slider {
  position: relative;
  margin-top: -20px;
}

/* 1 */

/* Override id swiper style */

.swiper-button-next,
.swiper-button-prev {
  display: flex;
  align-items: center;
  justify-content: space-around;
  pointer-events: none;

  width: 200px;
  height: 400px;
  background-image: none;
  background: radial-gradient(
    rgba(255, 255, 255, 0.85) 25%,
    rgba(255, 255, 255, 0.75) 35%,
    rgba(255, 255, 255, 0.4) 50%,
    rgba(255, 255, 255, 0) 65%
  );
}

.swiper-button-next {
  right: 0px;
  transform: translate(35%, -60%);
}

.swiper-button-prev {
  left: 0px;
  transform: translate(-35%, -60%);
}

.swiper-button-next i,
.swiper-button-prev i {
  font-size: 3.5rem;
  padding: 1.5rem;
  pointer-events: auto;
}

.swiper-container-horizontal > .swiper-pagination-bullets {
  top: 10px;
  text-align: right;
  height: 20px;
}

.swiper-pagination-bullet {
  width: 14px;
  height: 14px;
}

.swiper-pagination-bullet-active {
  background: #00303b;
}

.swiper-pagination-show {
  display: block;
}
.swiper-pagination-hide {
  display: none;
}

.mobile-slider-wrapper {
  overflow-x: scroll;
  overflow-y: hidden;

  -webkit-overflow-scrolling: touch;
}

.mobile-slider-wrapper::-webkit-scrollbar {
  display: none;
}

.mobile-slider {
  white-space: nowrap;
  -webkit-backface-visibility: hidden;
  -webkit-transform: translate3d(0, 0, 0);
}

.mobile-slider > div {
  white-space: normal;
}

@media only screen and (min-width: 480px) {
  .belt .header .list-card-nohover h1 {
    padding-bottom: 10px;
    padding-top: 10px;
  }
}
