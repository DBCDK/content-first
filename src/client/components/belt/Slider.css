.belts {
  padding-right: 30px;
}

.desktop-slider {
  position: relative;
  margin-top: -20px;
}
/* Override id swiper style */
.swiper-button-next,
.swiper-button-prev {
  font-size: 3rem;
  background-image: none;
  position: absolute !important;
}
.swiper-button-next {
  right: -30px;
}
.swiper-button-prev {
  left: -30px;
}

.swiper-container {
  padding-top: 40px;
}

.swiper-container-horizontal > .swiper-pagination-bullets {
  top: 0px;
  text-align: right;
  height: 20px;
}

.swiper-pagination-bullet {
  width: 14px;
  height: 14px;
}
.swiper-pagination-bullet-active {
  background: #00303b;
}

.mobile-slider-wrapper {
  overflow-x: scroll;
  overflow-y: hidden;

  -webkit-overflow-scrolling: touch;
}
.mobile-slider-wrapper::-webkit-scrollbar {
  display: none;
}
.mobile-slider {
  white-space: nowrap;
  -webkit-backface-visibility: hidden;
  -webkit-transform: translate3d(0, 0, 0);
}

.mobile-slider > div {
  white-space: normal;
}

@media only screen and (min-width: 480px) {
  .belt .header .list-card-nohover h1{
    padding-bottom: 10px;
    padding-top: 10px;
  }
}