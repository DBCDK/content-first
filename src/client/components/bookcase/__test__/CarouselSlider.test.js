import React from 'react';
import renderer from 'react-test-renderer';

import Slider from '../CarouselSlider.component';

describe('Slider', () => {
  it('renders initial component', () => {
    const tree = renderer
      .create(
        <Slider slideIndex={0} onNextBook={''}>
          <h1 key="1">test</h1>
          <h1 key="2">test</h1>
        </Slider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
