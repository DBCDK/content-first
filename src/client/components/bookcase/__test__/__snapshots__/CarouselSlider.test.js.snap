// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders initial component 1`] = `
<react-slick
  afterChange={[Function]}
  arrows={true}
  dots={true}
  dotsClass="slick-dots slick-dots-carousel"
  draggable={true}
  infinite={true}
  initialSlide={0}
  nextArrow={
    <NextArrow
      className="slick-next"
    />
  }
  prevArrow={
    <PrevArrow
      className="slick-prev"
    />
  }
  speed={500}
  variableWidth={false}
>
  <div
    className="slide-wrapper"
    style={
      Object {
        "display": "inline-block",
        "padding": 20,
        "verticalAlign": "top",
      }
    }
  >
    <h1>
      test
    </h1>
  </div>
  <div
    className="slide-wrapper"
    style={
      Object {
        "display": "inline-block",
        "padding": 20,
        "verticalAlign": "top",
      }
    }
  >
    <h1>
      test
    </h1>
  </div>
</react-slick>
`;
